### ISSUE TITLE
[feature] Inheriting from a user's CMake preset

### ISSUE THREAD

**Issue by mpusz** [conan-io/conan#13580]
### What is your suggestion?

Right now, it is possible to [generate Conan presets and inherit your own from them](https://docs.conan.io/2/examples/tools/cmake/cmake_toolchain/extend_own_cmake_presets.html#examples-tools-cmake-toolchain-build-project-extend-presets). However the opposite is not possible.

With the usage of CMakePresets some IDEs stopped adding development-specific flags provided in their configuration. For example, [VSCode does not generate compile_commands.json anymore](https://github.com/microsoft/vscode-cmake-tools/issues/1835). Adding this one setting manually to every Conan-generated preset is really inconvenient. It would be great if the users could provide their own preset file with the base configuration like that:

```json
{
    "name": "base",
    "cacheVariables": {
        "CMAKE_EXPORT_COMPILE_COMMANDS": "ON"
    },
    "hidden": true
}
```

specify it on a `[conf]` section as a base preset file which would make Conan include that file and inherit every preset from it. This would be a great improvement that would allow users to inject their own development-specific configuration settings into the pipeline.

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
We have thought about this, but this has some significant challenges. Like for example, how would you map the user configuration to Conan created configurations? You just consider 1 setting, but if we open this door, users will want to include different settings to different configurations, and it will probably be a nightmare of UX.

**Comment by mpusz** [comment]
Yeah, I see what you mean. You could hardcode some patterns for such use cases.

As a workaround, we can use a custom toolchain file instead, but I am not sure if all such development environment settings naturally belong to a toolchain file.
